src/Aspire.Hosting/Dashboard/proto/resource_service.proto

```diff
@@ -0,0 +1,170 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+syntax = "proto3";
+
+package aspire.v1;
+
+import "google/protobuf/timestamp.proto";
+
+////////////////////////////////////////////
+
+message ApplicationInformationRequest {
+}
+
+message ApplicationInformationResponse {
+    string application_name = 1;
+    string application_version = 2;
+}
+
+////////////////////////////////////////////
+
+message ResourceCommandRequest {
+    string command_type = 1;
+    ResourceId resource_id = 2;
+    optional string confirmation_message = 3;
+}
+
+enum ResourceCommandResponseKind {
+    none = 0;
```

Comment:
What does "none" mean?

Reply:
none -> undefined?

Reply:
btw, enum values should be shouty case, e.g. NONE, UPDATE_RECORD, etc

Reply:
Protobuf needs a zero-valued enum member, and it feels like making it a non-member of the enum is idiomatic, from the (internal) examples I've seen.

Undefined might be a better name. Will update. I'll change the case as well.